test/parallel/test-string-decoder.js

```diff
@@ -193,11 +193,10 @@ function unicodeEscape(str) {
 //   [ [ 0, 1 ], [ 1, 3 ] ],
 //   [ [ 0, 1 ], [ 1, 2 ], [ 2, 3 ] ]
 // ]
-function writeSequences(length, start, sequence) {
-  if (start === undefined) {
-    start = 0;
-    sequence = [];
-  } else if (start === length) {
+function writeSequences(length, start = 0, sequence = []) {
+  if (!start)
```

Comment:
This means that `start = 0` just always outputs an empty array. It should only do that if `length === 0 & start === 0`.

The fact that we have to add all these checks kind of illustrates why I don't think this change is a good idea or improves the current code.